src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,138 @@
+    public By newsletterButtonLocator = By.xpath(footersocialXpath + "//button[@class='btn btn-orange']");
+
+    public BasePage(WebDriver driver) {
+        this.driver = driver;
+        int baseWaitInSeconds = Integer.parseInt(ConfigReader.getProperty(ConfigReader.PropertyKeys.BASE_WAIT_IN_SECONDS));
+        this.wait = new WebDriverWait(driver, Duration.ofSeconds(baseWaitInSeconds));
+        this.baseUrl = ConfigReader.getProperty(ConfigReader.PropertyKeys.BASE_URL);
+    }
+
+    public By buttonNavbarLocatorByDataId(NavbarButton button) {
+        return By.xpath(navbarXpath + "//li[@data-id='"+ button.getDataId() +"']//span[@class='menu_text']");
+    }
+
+    public By buttonCategoryMenuLocatorByText(CategoryMenuButton button) {
+        return By.xpath(categoryMenuXpath + "//a[contains(text(), '" + button.getText() + "')]");
+    }
+
+    public By hyperLinkFooterLocatorByText(FooterHyperLink link) {
+        return By.xpath(footerlinksXpath + "//a[text()='" + link.getText() + "']");
+    }
+
+    public void openUrl(String url) {
+        driver.get(url);
+    }
+
+    public void refresh() {
+        driver.navigate().refresh();
+    }
+
+    public void clickOnElement(By locator) {
+        locateElement(locator).click();
+    }
+
+    public void scrollToElement(By locator) {
+        WebElement element = locateElement(locator);
+        JavascriptExecutor js = (JavascriptExecutor) driver;
+        js.executeScript("arguments[0].scrollIntoView(true);", element);
+    }
+
+    public void hoverOverElement(By locator) {
+        WebElement element = locateElement(locator);
+        Actions actions = new Actions(driver);
+        actions.moveToElement(element).perform();
+    }
+
+    public void selectCheckbox(By locator) {
+        WebElement element = locateElement(locator);
+        if (!element.isSelected()) {
+            element.click();
+        }
+    }
+
+    public void deselectCheckbox(By locator) {
+        WebElement element = locateElement(locator);
+        if (element.isSelected()) {
+            element.click();
+        }
+    }
+
+    public void selectRadioButton(By locator) {
+        WebElement element = locateElement(locator);
+        if (!element.isSelected()) {
+            element.click();
+        }
+    }
+
+    public void setCheckboxState(By locator, boolean state) {
+        WebElement element = locateElement(locator);
+        if (element.isSelected() != state) {
+            element.click();
+        }
+    }
+
+    public void sendKeysToElement(By locator, String text) {
+        locateElement(locator).sendKeys(text);
+    }
+
+    public String getTextFromElement(By locator) {
+        return locateElement(locator).getText();
+    }
+
+    public String getAttributeFromElement(By locator, String attribute) {
+        return locateElement(locator).getAttribute(attribute);
+    }
+
+    public String getTagNameFromElement(By locator) {
```

Comment:
I bet it's not going to be used :D 
I understand the approach you took, but in my own projects I prefer not to add unused code so you might expect it's not going to be approved in other projects as well :) 

Pros of removing unused code:
https://stackoverflow.com/questions/15699995/could-someone-explain-the-pros-of-deleting-or-keeping-unused-code

Reply:
Good to know. I will now try not to write redundant code, thanks 👨‍🏫👨‍🏫